examples/options/include/traccc/options/generation.hpp

```diff
@@ -0,0 +1,63 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/options/details/interface.hpp"
+#include "traccc/options/details/value_array.hpp"
+
+// System include(s).
+#include <cstddef>
+
+namespace traccc::opts {
+
+/// Configuration for particle / event generation
+class generation : public interface {
+
+    public:
+    /// @name Options
+    /// @{
+
+    /// The number of events to generate
+    std::size_t events = 1;
+
+    /// The number of particles to generate per event
+    unsigned int gen_nparticles{1u};
+    /// Vertex position [mm]
+    opts::value_array<float, 3> vertex{0.f, 0.f, 0.f};
+    /// Standard deviation of the vertex position [mm]
+    opts::value_array<float, 3> vertex_stddev{0.f, 0.f, 0.f};
+    /// Range of momentum [GeV]
+    opts::value_array<float, 2> mom_range{1.f, 1.f};
```

Comment:
:thinking: The idea here is to allow the user to specify these values "in GeV". Because that tends to be more convenient than doing so in MeV. (Though I don't have a strong opinion about that part.) If we were to use `unit<float>::GeV` here, the user would have to use "native units" on the command line. Which would depend on how these unit definitions are defined exactly.

By saying that "these values are to be understood in GeV", and later multiplying them by `unit<float>::GeV`, we make the command line usage a little easier.

Now... one could do something a lot better. We could pick up (or even implement) a "units library" that does all of this in a better way. Where we don't just use floating point numbers multiplied by constants, but actual types. I.e. something like:

```c++
opts::value_array<traccc::energy, 2> mom_range{1_gev, 1_gev};
```

I've been seeing https://github.com/aurora-opensource/au advertised at CppCon for many years now. :thinking: We could consider picking it up...

But for now I'd stay with this current, slightly fragile setup.

Reply:
Ahh, wait... That is actually a fair point. There is indeed no multiplication by `GeV` here:

https://github.com/acts-project/traccc/blob/main/examples/options/src/options/particle_gen_options.cpp#L58-L63

:thinking: @beomki-yeo, did Joana just find a bug? Since these values don't seem to be multiplied later on either...
  - https://github.com/acts-project/traccc/blob/main/examples/simulation/simulate_telescope.cpp#L108
  - https://github.com/acts-project/traccc/blob/main/examples/simulation/simulate.cpp#L114
  - https://github.com/acts-project/traccc/blob/main/examples/simulation/simulate_toy_detector.cpp#L82
  - https://github.com/acts-project/traccc/blob/main/examples/simulation/simulate_wire_chamber.cpp#L83

And yes, I absolutely want to get rid of these "configuration conversions" in a future PR. Such that `traccc::opts::generation` would have a generation configuration object directly. We'll just need to make sure that the same as how it works for some other configuration types, it could be used without needing the full templated type of `detray::random_track_generator`. :thinking:

Reply:
No, I think GeV is the native unit in detray. It should not change anything, just make it more robust. The phi angles in degrees, I am not so sure about

Reply:
The track generator probably clamps that to [-pi, pi], I would guess

Reply:
The (phi) angles are treated correctly. They **are** being multiplied by `detray::unit<float>::degree`.

Okay, so no actual bug here. I'll still add a multiplication by `GeV` to make the code a bit more robust. :thinking:

Reply:
Ah, OK, all good then